File: go/send_sms.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
	"os"
)

func main() {
	key := os.Getenv("SMSROUTE_API_KEY")
	if key == "" {
		fmt.Fprintln(os.Stderr, "Set SMSROUTE_API_KEY")
		os.Exit(1)
	}
	to := "+14155550123"
	if len(os.Args) > 1 {
		to = os.Args[1]
	}
	payload, _ := json.Marshal(map[string]string{
		"to":   to,
		"from": "INFO",
		"body": "Hello from SMSRoute examples",
	})
	req, err := http.NewRequest("POST", "https://api.smsroute.cc/sms/send", bytes.NewReader(payload))
	if err != nil {
		panic(err)
	}
	req.Header.Set("Authorization", "Bearer "+key)
	req.Header.Set("Content-Type", "application/json")
	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	buf := new(bytes.Buffer)
	buf.ReadFrom(resp.Body)
	fmt.Println(resp.StatusCode, buf.String())
}

File: php/send_sms.php
<?php
// php send_sms.php +14155550123 "Hello"
$key = getenv("SMSROUTE_API_KEY");
if (!$key) {
    fwrite(STDERR, "Set SMSROUTE_API_KEY\n");
    exit(1);
}
$to = $argv[1] ?? "+14155550123";
$body = $argv[2] ?? "Hello from SMSRoute examples";

$ch = curl_init("https://api.smsroute.cc/sms/send");
curl_setopt_array($ch, [
    CURLOPT_POST => true,
    CURLOPT_HTTPHEADER => [
        "Authorization: Bearer " . $key,
        "Content-Type: application/json",
    ],
    CURLOPT_POSTFIELDS => json_encode([
        "to" => $to,
        "from" => "INFO",
        "body" => $body,
    ]),
    CURLOPT_RETURNTRANSFER => true,
]);
$resp = curl_exec($ch);
$code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);
echo $code . " " . $resp . PHP_EOL;
